quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/KustoSampleApp.java

```diff
@@ -0,0 +1,395 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package com.microsoft.azure.kusto.quickstart;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.kusto.data.*;
+import com.microsoft.azure.kusto.data.auth.ConnectionStringBuilder;
+import com.microsoft.azure.kusto.data.exceptions.DataClientException;
+import com.microsoft.azure.kusto.data.exceptions.DataServiceException;
+import com.microsoft.azure.kusto.ingest.*;
+import com.microsoft.azure.kusto.ingest.exceptions.IngestionClientException;
+import com.microsoft.azure.kusto.ingest.exceptions.IngestionServiceException;
+import com.microsoft.azure.kusto.ingest.source.BlobSourceInfo;
+import com.microsoft.azure.kusto.ingest.source.FileSourceInfo;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.security.GeneralSecurityException;
+import java.security.PrivateKey;
+import java.security.cert.X509Certificate;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+import static com.microsoft.azure.kusto.data.ClientRequestProperties.OPTION_SERVER_TIMEOUT;
+
+public class KustoSampleApp {
+    // TODO - Config:
+    //  If this quickstart app was downloaded from OneClick, kusto-sample-config.json should be pre-populated with your cluster's details
+    //  If this quickstart app was downloaded from GitHub, edit kusto-sample-config.json and modify the cluster URL and database fields appropriately
+    private static final String CONFIG_FILE_NAME = "kusto-sample-config.json";
+
+    // TODO - Config (Optional): Change the authentication method from User Prompt to any of the other options
+    //  Some of the auth modes require additional environment variables to be set in order to work (see usage in generateConnectionString below)
+    //  Managed Identity Authentication only works when running as an Azure service (webapp, function, etc.)
+    private static final String AUTHENTICATION_MODE = "UserPrompt"; // Options: (UserPrompt|ManagedIdentity|AppKey|AppCertificate)

```

Comment:
DeviceCode?

Reply:
That's on my plate, but I didn't want it to hold up getting this PR out.